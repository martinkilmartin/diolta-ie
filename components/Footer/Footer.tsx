import { UListLinks } from '../ListLinks'
import { Copyright } from '../Copyright'

type Props = {
  links?: string[]
  brand: string
}

const Footer = ({ links, brand }: Props): JSX.Element => (
  <footer className="w-full max-w-screen-xl px-4 py-8 mx-auto bg-neutral text-neutral-content rounded-box">
    {links && <UListLinks links={links} />}
    <Copyright brand={brand} />
  </footer>
)

export default Footer
