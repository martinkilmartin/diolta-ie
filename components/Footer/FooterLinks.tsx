import Link from 'next/link'

type Props = {
  links: {
    title: string
    urls: {
      path: string
      text: string
    }[]
  }[]
}

const FooterLinks = ({ links }: Props): JSX.Element => (
  <div className="grid-rows footer">
    {links.map((link, i) => (
      <div key={i}>
        <span className="footer-title">{link.title}</span>
        {link.urls.map((link, index) => (
          <Link key={index} href={link.path}>
            <a className="link link-hover">{link.text}</a>
          </Link>
        ))}
      </div>
    ))}
  </div>
)

export default FooterLinks
