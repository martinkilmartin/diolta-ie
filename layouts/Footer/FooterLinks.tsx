type Props = {
  links: {
    title: string
    urls: {
      path: string
      text: string
    }[]
  }[]
}

const FooterLinks = ({ links }: Props): JSX.Element => (
  <div className="grid-rows footer">
    {links.map((link, i) => (
      <div key={i}>
        <span className="opacity-100 footer-title">{link.title}</span>
        {link.urls.map((link, index) => (
          <a key={index} href={link.path} className="link link-hover">
            {link.text}
          </a>
        ))}
      </div>
    ))}
  </div>
)

export default FooterLinks
