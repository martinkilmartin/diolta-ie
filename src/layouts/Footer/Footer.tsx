import { FooterLinks } from '@layouts/Footer'
import { Copyright } from '@components/Copyright'

import { FOOTER_LINKS } from '@constants/FOOTER_LINKS'

type Props = {
  brand: string
}

const Footer = ({ brand }: Props): JSX.Element => (
  <footer className="p-10 footer bg-primary text-primary-content footer-center">
    <FooterLinks links={FOOTER_LINKS} />
    <div>
      <Copyright brand={brand} />
    </div>
  </footer>
)

export default Footer
