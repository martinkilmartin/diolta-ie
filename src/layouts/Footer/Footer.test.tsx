import renderer from 'react-test-renderer'
import Footer from './Footer'

test('Footer displays copyright with current year and brand', () => {
  const brand = 'Acme Teo.'
  const footerRenderer = renderer.create(<Footer brand={brand} />)
  const tree = footerRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const footerInstance = footerRenderer.root
  const footerElement = footerInstance.findByProps({
    className: 'p-10 footer bg-primary text-primary-content footer-center',
  })
  expect(footerElement.findByType('p').children[0]).toEqual(
    `Â© ${new Date().getFullYear()} ${brand}`
  )
})
