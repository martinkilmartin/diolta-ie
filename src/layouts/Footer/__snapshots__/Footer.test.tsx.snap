// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer displays copyright with current year and brand 1`] = `
<footer
  className="p-10 footer bg-primary text-primary-content footer-center"
>
  <div
    className="grid-rows footer"
  >
    <div>
      <span
        className="opacity-100 footer-title"
      >
        Company
      </span>
      <a
        className="link link-hover"
        href="/contact"
      >
        Contact
      </a>
    </div>
    <div>
      <span
        className="opacity-100 footer-title"
      >
        Legal
      </span>
      <a
        className="link link-hover"
        href="/cookie"
      >
        Cookie Policy
      </a>
    </div>
  </div>
  <div>
    <p>
      Â© 2021 Acme Teo.
    </p>
  </div>
</footer>
`;
